apps/rig_inbound_gateway/lib/rig_inbound_gateway_web/session.ex

```diff
@@ -0,0 +1,226 @@
+defmodule RigInboundGatewayWeb.Session do
+  @moduledoc """
+  Maintains subscriptions and buffers incoming events.
+
+  Started by, or connected from, a (longpolling) connection process.
+  """
+  use GenServer
+
+  require Logger
+
+  alias Rig.EventFilter
+  alias RigCloudEvents.CloudEvent
+  alias RigInboundGatewayWeb.EventBuffer
+  alias RIG.Subscriptions
+  alias RigInboundGateway.Events
+
+  @window_size_ms 500
```

Comment:
Imo some of these values should be configurable via env vars. Question is, which might be interesting -- to not end up with crazy amount of configurations.

Reply:
@kevinbader your opinion?

Reply:
Imo, at least `event_buffer_size ` and `window_size_ms` should be configurable via env vars. When I think about it as a user/consumer, I would probably change those 2 according to my needs.

Reply:
I agree with `event_buffer_size` but not with `window_size_ms`. The latter is an internal thing, basically the polling interval for checking whether there are events available. Btw, perhaps it would be a good idea to rename that accordingly - it's not related to [what I would consider a window](https://en.wikipedia.org/wiki/Sliding_window_protocol) in this context.

Reply:
resolving in favor of #236